### PR TITLE
Use DataFlow relationships instead of LineageMappings for capturing lineage

### PR DESCRIPTION
**Pull request by popa-raluca**
## Description

Use DataFlow relationships instead of LineageMappings for capturing lineage. This PR contains changes for the following components involved in capturing and retrieving lineage: DE OMAS, AL OMAS, OLS, DS Connector

## Testing

Tested locally with updated FVTs, unit tests & DE OMAS samples

### CODE REVIEW

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java [hunk 1]
```diff
@@ -150,60 +152,66 @@ public Optional<EntityDetail> findSchemaTypeEntity(String userId, String qualifi
...
-                LINEAGE_MAPPING_TYPE_NAME, sourceEntity.get().getType().getTypeDefName(),
-                targetEntity.get().getType().getTypeDefName(), externalSourceName, null);
+
+        InstanceProperties relationshipProperties = repositoryHelper.addStringPropertyToInstance(serviceName, null,
+                OpenMetadataAPIMapper.DESCRIPTION_PROPERTY_NAME, description, methodName);
+        relationshipProperties = repositoryHelper.addStringPropertyToInstance(serviceName, relationshipProperties,
+                OpenMetadataAPIMapper.FORMULA_PROPERTY_NAME, formula, methodName);
+
+        dataEngineCommonHandler.upsertExternalRelationship(userId, dataSupplierEntity.get().getGUID(), dataconsumertEntity.get().getGUID(),
+                DATA_FLOW_TYPE_NAME, dataSupplierEntity.get().getType().getTypeDefName(),
```
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java:195]: False positive, an exception is thrown if the Optional is empty

**Review by alexandra-bucur — APPROVED** [review]

File: open-metadata-implementation/governance-servers/data-engine-proxy-services/data-engine-proxy-connector/src/main/java/org/odpi/openmetadata/governanceservers/dataengineproxy/connectors/DataEngineInterface.java [hunk 2]
```diff
@@ -105,12 +105,12 @@
...
-     * Retrieve a list of the changed lineage mappings between the dates and times provided.
+     * Retrieve a list of the changed data flows between the dates and times provided.
      *
      * @param from the date and time from which to look for changes (exclusive)
      * @param to the date and time up to which to look for changes (inclusive)
-     * @return {@code List<LineageMapping>}
+     * @return {@code List<DataFlow>}
      */
-    List<LineageMapping> getChangedLineageMappings(Date from, Date to) throws ConnectorCheckedException, PropertyServerException;
+    List<DataFlow> getChangedDataFlows(Date from, Date to) throws ConnectorCheckedException, PropertyServerException;
```
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/governance-servers/data-engine-proxy-services/data-engine-proxy-connector/src/main/java/org/odpi/openmetadata/governanceservers/dataengineproxy/connectors/DataEngineInterface.java:114]: False positive, the implementation classes are using it

File: open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/handlers/DataEngineSchemaTypeHandler.java:195]: @sonatype-lift ignore

File: open-metadata-implementation/access-services/data-engine/data-engine-client/src/main/java/org/odpi/openmetadata/accessservices/dataengine/client/DataEngineRESTClient.java [hunk 3]
```diff
@@ -326,18 +326,18 @@ public String addProcessHierarchy(String userId, ProcessHierarchy processHierarc
...
+    public void addDataFlows(String userId, List<DataFlow> dataFlows) throws InvalidParameterException,
+                                                                             UserNotAuthorizedException,
+                                                                             PropertyServerException {
+        final String methodName = DATA_FLOWS_METHOD_NAME;
 
         invalidParameterHandler.validateUserId(userId, methodName);
 
-        LineageMappingsRequestBody requestBody = new LineageMappingsRequestBody();
-        requestBody.setLineageMappings(lineageMappings);
+        DataFlowsRequestBody requestBody = new DataFlowsRequestBody();
```
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/access-services/data-engine/data-engine-client/src/main/java/org/odpi/openmetadata/accessservices/dataengine/client/DataEngineRESTClient.java:336]: @sonatype-lift ignore

File: open-metadata-implementation/access-services/data-engine/data-engine-client/src/main/java/org/odpi/openmetadata/accessservices/dataengine/client/DataEngineEventClient.java [hunk 4]
```diff
@@ -260,13 +260,13 @@ public String addProcessHierarchy(String userId, ProcessHierarchy processHierarc
      * @throws ConnectorCheckedException problem with the underlying connector (if used)
      */
     @Override
-    public void addLineageMappings(String userId, List<LineageMapping> lineageMappings) throws InvalidParameterException, ConnectorCheckedException {
+    public void addDataFlows(String userId, List<DataFlow> dataFlows) throws InvalidParameterException, ConnectorCheckedException {
 
-        LineageMappingsEvent event = new LineageMappingsEvent();
+        DataFlowsEvent event = new DataFlowsEvent();
```
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/access-services/data-engine/data-engine-client/src/main/java/org/odpi/openmetadata/accessservices/dataengine/client/DataEngineEventClient.java:265]: @sonatype-lift ignore

File: open-metadata-implementation/governance-servers/data-engine-proxy-services/data-engine-proxy-services-server/src/main/java/org/odpi/openmetadata/governanceservers/dataengineproxy/processor/DataEngineProxyService.java [hunk 5]
```diff
@@ -310,25 +310,25 @@ private void upsertProcessHierarchies(Date changesLastSynced,
...
-                                       Date changesCutoff) throws
+    private void upsertDataFlows(Date changesLastSynced,
+                                 Date changesCutoff) throws
             InvalidParameterException,
             PropertyServerException,
             UserNotAuthorizedException,
             ConnectorCheckedException {
-        final String methodName = "upsertLineageMappings";
-        final String type = "LineageMappings";
+        final String methodName = "upsertDataFlows";
```
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/governance-servers/data-engine-proxy-services/data-engine-proxy-services-server/src/main/java/org/odpi/openmetadata/governanceservers/dataengineproxy/processor/DataEngineProxyService.java]: @sonatype-lift ignore

File: open-metadata-implementation/governance-servers/data-engine-proxy-services/data-engine-proxy-connector/src/main/java/org/odpi/openmetadata/governanceservers/dataengineproxy/connectors/DataEngineInterface.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/governance-servers/data-engine-proxy-services/data-engine-proxy-connector/src/main/java/org/odpi/openmetadata/governanceservers/dataengineproxy/connectors/DataEngineInterface.java:114]: @sonatype-lift ignore

File: open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/DataFlow.java [hunk 6]
```diff
@@ -25,30 +25,34 @@
 @Setter
```
[the remark this answers is not in the document]
Comment (popa-raluca) [reply open-metadata-implementation/access-services/data-engine/data-engine-api/src/main/java/org/odpi/openmetadata/accessservices/dataengine/model/DataFlow.java:25]: @sonatype-lift ignore

**Pull request merged**